packages/backend-common/src/s3.ts

```diff
@@ -174,6 +177,44 @@ export const getObjectWithPresignedUrl = async (
 	return destinationPath;
 };
 
+export const uploadObjectWithPresignedUrl = async (
+	presignedUrl: string,
+	metadata: MediaMetadata,
+) => {
+	logger.info(
+		`Uploading with ${presignedUrl}, metadata ${JSON.stringify(metadata)}`,
+	);
+	const fileStream = createReadStream(metadata.mediaPath);
+
+	try {
+		const webStream = Readable.toWeb(fileStream) as ReadableStream;
+		const fileStats = await stat(metadata.mediaPath);
+		const contentLength = fileStats.size;
+		const response = await fetch(presignedUrl, {
+			method: 'PUT',
+			body: webStream,
+			duplex: 'half',
+			headers: {
+				'Content-Type': 'application/octet-stream',
+				'Content-Length': contentLength.toString(),
+			},
+		} as RequestInit);
+
+		if (response.ok) {
+			const text = await response.text();
+			logger.info(`File uploaded successfully, response: ${text}`);
+			return true;
```

Comment:
[minor] this function never returns false (only returns true or it throws) so perhaps this return is redundant